### ISSUE TITLE
Inspector incorrectly points out that #[derive] can only be used on structs, enums and unions

### ISSUE THREAD

**Issue by hamza1311** [intellij-rust/intellij-rust#6162]
## Environment

* **IntelliJ Rust plugin version:** `0.3.131.3366-202`
* **Rust toolchain version:** `rustc 1.46.0 (04488afe3 2020-08-24)`
* **IDE name and version:** `IntelliJ IDEA 2020.2.2 (Ultimate Edition), Build #IU-202.7319.50`
* **Operating system:** `Linux, Fedora 32`

## Problem description
When I apply `#[derive]`on a `type` inspector incorrectly points out that `#[derive]` can only be used on structs, enums and unions as shown by the following screenshot:
![image](https://user-images.githubusercontent.com/47357913/94343288-8aadac80-0030-11eb-89df-1b279b706e70.png)

The code in question here is: 
```rust
#[wasm_bindgen(module = "/build/built-js.js")]
extern "C" {
    #[derive(Debug)]
    type IconButtonToggle;

    #[wasm_bindgen(getter, static_method_of = IconButtonToggle)]
    fn _dummy_loader() -> JsValue;
}
```

## Steps to reproduce

- Apply `#[derive]` to a `type`

I don't know how much code I need to provide so I will link the repository [here](https://github.com/hamza1311/yew-material-components). This error is shown in files under `src/` directory (excluding `lib.rs`). The code snippet given above is from [here](https://github.com/hamza1311/yew-material-components/blob/68f611769ee47f209f5e842b6b51f954967c6b78/src/icon_button_toggle.rs#L5-L12).

**Comment by Stzx** [comment]
I checked the related feature links and other issues. I didn't find any documents or instructions that can use `derive` on `type` defined in `extern`.

Can you provide some official standards?

**Comment by Undin** [comment]
@Stzx I suppose `derive` attribute is handled by proc macro here

**Comment by Stzx** [comment]
> @Stzx I suppose `derive` attribute is handled by proc macro here

Yes, but I couldn't find a definition for it. Maybe I'm missing something.

**Comment by hamza1311** [comment]
Correction: after some testing, it appears that this is a `wasm-bindgen` thing. `cargo expand`ing shows that `wasm-bindgen` generates a `struct` for the `type` defined in `extern "C"` blocks. Those types can have proc macros as shown in examples [here](https://rustwasm.github.io/wasm-bindgen/reference/attributes/on-js-imports/extends.html) and hence `#[derive(...)]` works fine with it. [Here's](https://pastify-app.web.app/show/JKxAH0wf4AG1wFpInhaP) the relevant `cargo expand` output if that matters to anyone. Removing `#[wasm_bindgen]` from `extern "C"` results in the expected compilation error of ``` `derive` may only be applied to structs, enums and unions```

---

I will leave this issue open for now. Feel free to close this if this isn't a something that should be fixed, although I don't think special casing something for `wasm-bindgen` would unreasonable.

**Comment by jerryc05** [comment]
I also encountered this problem. I assume that we have to fix/improve the macro expansion engine before solving this bug. 
If intellirust can "foresee" that `#[wasm_bindgen]` will generate a struct for the type, this error will probably automatically disappear. To dig deeper into this problem, I think the root of this bug could be intellirust's failing to expand `#[proc_macro_attribute]` because `#[wasm_bindgen]` failed to display its declaration when I `ctrl-click`ed it.
